src/Process/LayeredMaterialNodeProcessing.js

```diff
@@ -9,30 +9,48 @@ import OGCWebServiceHelper, { SIZE_TEXTURE_TILE } from '../Core/Scheduler/Provid
 function initNodeImageryTexturesFromParent(node, parent, layer) {
     if (parent.materials && parent.materials[RendererConstant.FINAL].getColorLayerLevelById(layer.id) > EMPTY_TEXTURE_ZOOM) {
         const coords = node.getCoordsForLayer(layer);
-        const offsetTextures = node.materials[RendererConstant.FINAL].getLayerTextureOffset(layer.id);
 
-        let textureIndex = offsetTextures;
+        const indexInNode = node.material.indexOfColorLayer(layer.id);
+        const indexInParent = parent.material.indexOfColorLayer(layer.id);
+
+        const offsetScale = node.materials[RendererConstant.FINAL]
+            .offsetScale[l_COLOR][layer.id];
+
+        let found = 0;
+        const atlas = parent.material.uniforms.atlasTextures.value[indexInParent];
         for (const c of coords) {
-            for (const texture of parent.materials[RendererConstant.FINAL].getLayerTextures(l_COLOR, layer.id)) {
-                if (c.isInside(texture.coords)) {
-                    const result = c.offsetToParent(texture.coords);
-                    node.material.textures[l_COLOR][textureIndex] = texture;
-                    node.material.offsetScale[l_COLOR][textureIndex] = result;
-                    textureIndex++;
+            for (let i = 0; i < atlas.coords.length; i++) {
+                const parentCoords = atlas.coords[i];
+
+                if (c.isInside(parentCoords)) {
```

Comment:
(you're commenting on existing code that I simply updated)

And I *think* a PM texture will always be fully contained in its parent textures which are PM as well (would need to double-check this though)